client/order/index.js

```diff
@@ -42,30 +44,53 @@ jQuery( function ( $ ) {
 			'wc-refunded' === this.value &&
 			'wc-refunded' !== originalStatus
 		) {
-			if ( ! canRefund ) {
-				alert(
-					__( 'Order cannot be refunded', 'woocommerce-payments' )
-				);
-				return;
-			}
-			if ( 0 >= refundAmount ) {
-				alert( __( 'Invalid Refund Amount', 'woocommerce-payments' ) );
+			renderRefundConfirmationModal(
+				originalStatus,
+				canRefund,
+				refundAmount
+			);
+		} else if (
+			'wc-cancelled' === this.value &&
+			'wc-cancelled' !== originalStatus
+		) {
+			if ( ! canRefund || 0 >= refundAmount ) {
 				return;
 			}
-			const container = document.createElement( 'div' );
-			container.id = 'wcpay-refund-confirm-container';
-			document.body.appendChild( container );
-			ReactDOM.render(
-				<RefundConfirmationModal
-					orderStatus={ originalStatus }
-					refundAmount={ refundAmount }
-					formattedRefundAmount={ getConfig(
-						'formattedRefundAmount'
-					) }
-					refundedAmount={ getConfig( 'refundedAmount' ) }
-				/>,
-				container
+			renderModal(
+				<CancelConfirmationModal
+					originalOrderStatus={ originalStatus }
+				/>
```

Comment:
Since this flow has been refactored, adding tests for the cases covered here would be nice :) This would increase the code coverage and confidence for future refactors.

Reply:
Done